controllers/hotel_controller.go

```diff
@@ -0,0 +1,85 @@
+package controllers
+
+import (
+	cError "github.com/couchbase-examples/golang-quickstart/errors"
+	"github.com/couchbase-examples/golang-quickstart/models"
+	services "github.com/couchbase-examples/golang-quickstart/service"
+	"github.com/gin-gonic/gin"
+	"net/http"
+)
+
+type HotelController struct {
+	HotelService services.IHotelService
+}
+
+func NewHotelController(airlineService services.IHotelService) *HotelController {
+	return &HotelController{
+		HotelService: airlineService,
+	}
+}
+
+// SearchByName
+//
+// @Summary Search by hotel name
+// @Description Search for hotels based on their name.
+// @Tags Hotel
+// @Produce json
+// @Param name query string true "name search"
```

Comment:
Can u please add an example like in other routes

Reply:
done